backendImpl/src/main/java/com/virtuslab/gitmachete/backend/impl/GitMacheteRootBranch.java

```diff
@@ -22,13 +25,17 @@
   private final List<GitMacheteNonRootBranch> downstreamBranches;
   private final IGitMacheteCommit pointedCommit;
   private final SyncToRemoteStatus syncToRemoteStatus;
+  @Getter(AccessLevel.NONE)
+  @Nullable
+  private final IGitMacheteRemoteBranch remoteBranch;
   @Nullable
   private final String customAnnotation;
 
   public GitMacheteRootBranch(String name, List<GitMacheteNonRootBranch> downstreamBranches,
       IGitMacheteCommit pointedCommit,
       SyncToRemoteStatus syncToRemoteStatus,
-      @Nullable String customAnnotation) {
+      @Nullable String customAnnotation,
+      @Nullable IGitMacheteRemoteBranch remoteBranch) {
```

Comment:
Maybe move up, to right before `syncToRemoteStatus`? To keep related params close to each other